.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```

Comment:
Why not run the tests when pushing to a new branch without an open pull request?
And why only run the tests on pull requests into master?

Reply:
Why should we?

If you push to a PR'd branch it is run every time and can be used to properly review and approve changes.

All branch pushes and PRs not going into master are not of interest and can be run during development locally. (Think green, safe resources :D)

What would be your reasoning to do it every time?

Reply:
Yeah saving resources makes sense. I just thought about a more flexible setup which ensures the tests are always executed regardless of the branch and pull request. GitHub makes it too cheap running too many tests.